backend/src/main/java/atlantbh/restaurants/controllers/RestaurantController.java

```diff
@@ -0,0 +1,57 @@
+package atlantbh.restaurants.controllers;
+
+import atlantbh.restaurants.models.Restaurant;
+import atlantbh.restaurants.models.filters.RestaurantFilterBuilder;
+import atlantbh.restaurants.models.sortkeys.RestaurantSortKeys;
+import atlantbh.restaurants.services.RestaurantService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class RestaurantController extends BaseController<Restaurant, RestaurantService> {
+
+    public ResponseEntity filter(@RequestParam(value = "q", required = false) String query,
+                                 @RequestParam(value = "priceRange", required = false) Integer priceRange,
+                                 @RequestParam(value = "pageSize", required = true) Integer pageSize,
+                                 @RequestParam(value = "pageNumber", required = true) Integer pageNumber,
+                                 @RequestParam(value = "sortKey", required = false) String sortKey,
+                                 @RequestParam(value = "name", required = false) String name,
+                                 @RequestParam(value = "category",required = false) String category,
+                                 @RequestParam(value = "cousine", required = false) String cousine,
+                                 @RequestParam(value = "location", required = false) String location){
+        RestaurantFilterBuilder rfb = new RestaurantFilterBuilder().setPriceRange(priceRange)
+                                                                    .setQuery(query)
+                                                                    .setName(name)
+                                                                    .setCategory(category)
+                                                                    .setCousine(cousine)
+                                                                    .setLocation(location)
+                                                                    .setPageSize(pageSize)
+                                                                    .setPageNumber(pageNumber);
+        if(sortKey != null)
```

Comment:
this setter should be also above, like every other..

Reply:
If sortKey is null then RestaurantSortKeys.valueOf(sortKey) will throw NullPointerException. That's the reason why i seperated this setter from others.